cmd/operator-sdk/generate/csv.go

```diff
@@ -74,76 +72,63 @@ Configure CSV generation by writing a config file 'deploy/olm-catalog/csv-config
+		"Paths to include in CSV generation, ex. \"deploy/prod,deploy/test\". "+
+			"If this flag is set and you want to enable default behavior, "+
+			"you must include \"deploy/\" in the argument list")
+	cmd.Flags().BoolVar(&c.updateCRDs, "update-crds", false,
+		"Update CRD manifests in deploy/{operator-name}/{csv-version} the using latest API's")
+	cmd.Flags().StringVar(&c.operatorName, "operator-name", "",
+		"Operator name to use while generating CSV")
+	cmd.Flags().StringVar(&c.csvChannel, "csv-channel", "",
+		"Channel the CSV should be registered under in the package manifest")
+	cmd.Flags().BoolVar(&c.defaultChannel, "default-channel", false,
+		"Use the channel passed to --csv-channel as the package manifests' default channel. "+
+			"Only valid when --csv-channel is set")
 
 	return cmd
 }
 
 func (c csvCmd) run() error {
 
-	absProjectPath := projutil.MustGetwd()
-	cfg := &input.Config{
-		AbsProjectPath: absProjectPath,
-		ProjectName:    filepath.Base(absProjectPath),
-	}
-	if projutil.IsOperatorGo() {
-		cfg.Repo = projutil.GetGoPkg()
-	}
-
 	log.Infof("Generating CSV manifest version %s", c.csvVersion)
 
-	csvCfg, err := catalog.GetCSVConfig(c.csvConfigPath)
-	if err != nil {
-		return err
-	}
 	if c.operatorName == "" {
-		// Use config operator name if not set by CLI, i.e. prefer CLI value over
-		// config value.
-		if c.operatorName = csvCfg.OperatorName; c.operatorName == "" {
-			// Default to using project name if both are empty.
-			c.operatorName = filepath.Base(absProjectPath)
-		}
+		c.operatorName = filepath.Base(projutil.MustGetwd())
 	}
-
-	s := &scaffold.Scaffold{}
-	csv := &catalog.CSV{
-		CSVVersion:     c.csvVersion,
-		FromVersion:    c.fromVersion,
-		ConfigFilePath: c.csvConfigPath,
-		OperatorName:   c.operatorName,
-	}
-	err = s.Execute(cfg, csv)
-	if err != nil {
-		return fmt.Errorf("catalog scaffold failed: %v", err)
+	cfg := gen.Config{
+		OperatorName: c.operatorName,
+		Filters:      gen.MakeFilters(c.includePaths...),
 	}
 
-	gcfg := gen.Config{
-		OperatorName: c.operatorName,
-		OutputDir:    filepath.Join(gencatalog.OLMCatalogDir, c.operatorName),
+	csv := gencatalog.NewCSV(cfg, c.csvVersion, c.fromVersion)
+	if err := csv.Generate(); err != nil {
+		return fmt.Errorf("error generating CSV: %v", err)
 	}
-	pkg := gencatalog.NewPackageManifest(gcfg, c.csvVersion, c.csvChannel, c.defaultChannel)
+	pkg := gencatalog.NewPackageManifest(cfg, c.csvVersion, c.csvChannel, c.defaultChannel)
 	if err := pkg.Generate(); err != nil {
 		return fmt.Errorf("error generating package manifest: %v", err)
 	}
 
 	// Write CRD's to the new or updated CSV package dir.
 	if c.updateCRDs {
-		input, err := csv.GetInput()
+		crdManifestSet, err := findCRDs(c.includePaths...)
 		if err != nil {
 			return err
 		}
-		err = writeCRDsToDir(csvCfg.CRDCRPaths, filepath.Dir(input.Path))
-		if err != nil {
-			return err
+		bundleDir := filepath.Join(gencatalog.OLMCatalogDir, strings.ToLower(c.operatorName), c.csvVersion)
```

Comment:
@estroz I might have missed this from previous PRs but if the CSV generator inputs are meant to be configurable i.e change [`g.inputs[DeployDirKey]`](https://github.com/operator-framework/operator-sdk/blob/5f3699c21f046fe411277338c04c855a505aa424/internal/generate/olm-catalog/csv.go#L52:2) to point to another location `deploy/production/` or in my case `config` for Kubebuilder's layout, then updating the CRD manifest to that new location would fail.
While the CSV and package manifests are written out to the new location, [`OLMCatalogDir`](https://github.com/operator-framework/operator-sdk/blob/5f3699c21f046fe411277338c04c855a505aa424/internal/generate/olm-catalog/csv.go#L48:2) is currently fixed to always be rooted at the `deploy/` directory so it will try to update the CRDs there.

Is there any way we can stop using that constant, and instead use the generator's input for `DeployDirKey`?

Not sure if any other cmds that need to have a configurable "deploy" directory are using this constant as well. That could be another thing to watch out for.

Reply:
Going to reconsider how inputs/filters are set up.